oak_sev_guest/src/ghcb.rs

```diff
@@ -636,6 +624,18 @@ where
         Ok(())
     }
 
+    /// Simple wrapper for operations that rely on the GHCB.
+    fn call<F>(&mut self, func: F) -> Result<(), &'static str>
```

Comment:
Alternatively you could call this `map` like in `Option` (and even allow the function to return something if the caller wants)